operator/helper/encoding.go

```diff
@@ -38,8 +39,37 @@ type EncodingConfig struct {
 }
 
 // Build will build an Encoding operator.
-func (c EncodingConfig) Build(context operator.BuildContext) (encoding.Encoding, error) {
-	return lookupEncoding(c.Encoding)
+func (c EncodingConfig) Build(context operator.BuildContext) (Encoding, error) {
+	enc, err := lookupEncoding(c.Encoding)
+	if err != nil {
+		return Encoding{}, err
+	}
+
+	return Encoding{
+		Encoding: enc,
+	}, nil
+}
+
+type Encoding struct {
+	Encoding encoding.Encoding
+}
+
+// decode converts the bytes in msgBuf to utf-8 from the configured encoding
+func (e *Encoding) Decode(msgBuf []byte) (string, error) {
+	decodeBuffer := make([]byte, 1<<12)
+	decoder := e.Encoding.NewDecoder()
```

Comment:
I create it every call to avoid race conditions
I left file reader implementation in file operator because it's optimized for that usecase

Reply:
Seems reasonable. If it proves to be a performance bottleneck at some point, we can look at optimizing then.